Frameworks/Foundation/_NSUndoGroup.mm

```diff
@@ -0,0 +1,93 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <Starboard.h>
+#import <Foundation/NSUndoManager.h>
+#import "_NSUndoManagerInternal.h"
+
+@implementation _NSUndoGroup {
+    StrongId<NSMutableArray*> _undoGrouping;
```

Comment:
Fair enough, using a list would be more appropriate then, with O(N) traversal (same as array) and O(1) insert/remove.  In normal operation, adding operations and performing undo will be far more frequent than removing specific target.

---
In reply to: [76125991](https://github.com/Microsoft/WinObjC/pull/900#discussion_r76125991) [](ancestors = 76125991,76124040)